internal/balancergroup/balancergroup.go

```diff
@@ -182,6 +182,29 @@ func (sbc *subBalancerWrapper) stopBalancer() {
 	sbc.balancer = nil
 }
 
+type balancerCache interface {
+	// Add adds item with key to the cache. callback is invoked when item is
+	// removed from the cache.
+	Add(key, item interface{}, callback func()) (interface{}, bool)
+
+	// Remove the item with the key from the cache.
+	Remove(key interface{}) (item interface{}, ok bool)
+
+	// Clear removes all entries, and runs the callbacks if runCallback is true.
+	Clear(runCallback bool)
+}
+
+// A no-op implementation of the balancerCache interface when caching is
+// disabled due to Options.SubBalancerCloseTimeout set to 0.
+type noopBalancerCache struct{}
```

Comment:
Do you think this should go into the cache package for other places to be able to share it?  And make the constructor for a timeout cache return it if the timeout is zero?

Reply:
I thought about this for a while. I feel that it does not make a whole lot of sense for a type that provides caching functionality to also provide a way to disable the same. 
- I feel that doing so would unnecessary add more complexity to the package providing the cache.
  - We would need to define an interface and create multiple implementations of it.
- Also, I don't see other usages of this cache where caching might have to be disabled. 
  - In fact, we could consider making the `cache.NewTimeoutCache` return an error if passed a timeout of `0`.

Instead explicitly handling the no-caching case in the `balancergroup` makes more sense because it makes things very clear to the reader of the code as to how and where caching is being done.

Reply:
I'm good with the current version.  I think if there _is_ a `nop` implementation then it makes the most sense in the cache package.  But just not calling the cache when it's disabled is a perfectly reasonable implementation too.